### PR TITLE
Add metrics: bridge_matrix_api_calls and …_failed

### PR DESCRIPTION
**Pull request by jaller94**
* Add the metrics `bridge_matrix_api_calls` and `bridge_matrix_api_calls_failed`

### CODE REVIEW

File: mautrix/client/api/events.py [hunk 1]
```diff
@@ -275,11 +279,13 @@ async def send_state_event(self, room_id: RoomID, event_type: EventType,
             https://matrix.org/docs/spec/client_server/r0.5.0#put-matrix-client-r0-rooms-roomid-state-eventtype-statekey
         """
         content = content.serialize() if isinstance(content, Serializable) else content
+        API_CALLS.labels(method="sendStateEvent").inc()
         resp = await self.api.request(Method.PUT, Path.rooms[room_id].state[event_type][state_key],
```
Comment (jaller94) [inline mautrix/client/api/events.py]: @Half-Shot Do I need to wrap this `request()` in a `try` block to catch network failures?

… or will `resp` just be undefined?

File: mautrix/client/api/events.py [hunk 2]
```diff
@@ -16,6 +16,10 @@
 from mautrix.types.event.state import state_event_content_map
 from .base import BaseClientAPI
 
+API_CALLS = Counter("bridge_matrix_api_calls",
+                    "The number of Matrix client API calls made", ("update_type",))
+API_CALLS_FAILED = Counter("bridge_matrix_api_calls_failed",
+                    "The number of Matrix client API calls which failed", ("update_type",))
```
Comment (jaller94) [inline mautrix/client/api/events.py]: Do I need to define the label `method`?

Comment (jaller94) [inline mautrix/client/api/events.py]: What's the meaning of "update_type"?

Comment (jaller94) [inline mautrix/client/api/events.py]: Do these need to be imported in a web endpoint function or does a creation of `Counter()` already add the metric to a Prometheus endpoint?

Comment (jaller94) [reply mautrix/client/api/events.py]: I assume this was the label.

File: mautrix/client/api/events.py (hunk 1 above)
Comment (Half-Shot) [reply mautrix/client/api/events.py]: I'd wrap in a try catch, but remember to just do `raise` so we allow it to fall through [like so](https://github.com/mautrix/signal/pull/164/files#diff-eea437b12b4747e9a300ff993e0194b85fe48c11b2b318410b72d5cbe63d2873R144-R151)

**Comment by jaller94** [comment]
Possible improvements (possibly in another PR, given the urgency):
- [ ] Cover other parts of the API, not just `events.py`.
- [ ] Write a wrapper function as the current code includes a lot of repetition.

File: mautrix/client/api/events.py [hunk 3]
```diff
@@ -14,8 +14,15 @@
...
                            ReactionEventContent, RelationType, Obj, Serializable)
 from mautrix.types.event.state import state_event_content_map
+from mautrix.util.opt_prometheus import Counter, Histogram
 from .base import BaseClientAPI
 
+API_CALLS = Counter("bridge_matrix_api_calls",
+                    "The number of Matrix client API calls made", ("method",))
+API_CALLS_FAILED = Counter("bridge_matrix_api_calls_failed",
+                           "The number of Matrix client API calls which failed", ("method",))
+API_CALLS_TIME = Histogram("bridge_matrix_api_calls_update",
```
Comment (jaller94) [inline mautrix/client/api/events.py]: Couldn't find a Histogram value like this on the NodeJS bridge repo, so I came up with `bridge_matrix_api_calls_update`.

`_update` is probably not a good name.

Comment (jaller94) [reply mautrix/client/api/events.py]: Renamed in https://github.com/vector-im/mautrix-python/pull/2/commits/6ab8cd5ad9556cac7897aefd008a0f0d745f7502

File: mautrix/client/api/events.py (hunk 2 above)
Comment (Half-Shot) [reply mautrix/client/api/events.py]: It's magic, you don't need to do anything :)

**Comment by Half-Shot** [comment]
> Write a wrapper function as the current code includes a lot of repetition.

I feel like there has gotta be a way to do this with attributes in python

**Review by Half-Shot — APPROVED** [review]
Code looks great, if it works we can shippit

**Comment by Half-Shot** [comment]
Ah, @jaller94 can you also open this upstream to mautrix as well?

**Pull request merged**